packages/rlc-codegen/src/helpers/camelCase.ts

```diff
@@ -0,0 +1,14 @@
+export function camelCase(
+  str: string,
+  options: { uppercaseThreshold?: number } = {}
+) {
+  const { uppercaseThreshold = 4 } = options;
+  const thresholdRegex = new RegExp(
+    `^(?<![A-Z])[A-Z]{1,${uppercaseThreshold}}(?![A-Z])`
+  );
+  if (!thresholdRegex.test(str)) {
+    return str;
+  }
+
+  return str.charAt(0).toLocaleLowerCase() + str.slice(1);
+}
```

Comment:
Should we combine camelCase.ts and pascalCase.ts to the same file such as nameUtils.ts ?